src/screens/SearchScreen.js

```diff
@@ -17,17 +17,26 @@ import {map, navKeys} from '../constants';
 
 class SearchScreen extends Component {
     componentWillMount() {
-        const {lastKnownRegion, initializeMap} = this.props;
+        const {lastKnownRegion} = this.props;
 
-        initializeMap(lastKnownRegion);
+        this.props.initializeMap(lastKnownRegion);
     }
 
     componentDidMount() {
-        const {lastKnownRegion, navigation: {setParams}, mapHasLoaded} = this.props;
+        const {lastKnownRegion, navigation: {setParams}} = this.props;
 
         setParams({buttonName: map.SearchOptions.LIST, toggleButton: this.toggleButton});
 
-        mapHasLoaded();
+        this.props.mapHasLoaded();
+    }
+
+    componentWillReceiveProps(nextProps){
+        const {sitesShouldUpdate, lastKnownRegion} = nextProps;
+
+        if(sitesShouldUpdate){
+            this.props.initializeMap(lastKnownRegion);
+        }
```

Comment:
This is the `sitesShouldUpdate` stuff I mentioned above. Whenever the `SearchScreen` gets a props update, if it should update the sites, then call the map initialization again. Then, when the initialization happens, it'll flip the `sitesShouldUpdate` flag back to false.